tests/test_settings.py

```diff
@@ -52,6 +52,29 @@ def test_set_setting_invalid_type():
         settings.set_setting("Threshold", "-30.0")
 
 
+def test_set_setting_invalid_type2():
+    """Test that the set_setting method raises a ValueError
+    when given an invalid type."""
+    settings = Settings()
+    with pytest.raises(ValueError):
+        settings.set_setting("voltage_base_mode", 1)
+
+
+def test_set_setting_invalid_type3():
+    """Test that the set_setting method raises a ValueError
+    when given an invalid type."""
+    settings = Settings()
+    with pytest.raises(ValueError):
+        settings.set_setting("max_spike_skip", 1.0)
```

Comment:
Here, we could just try to convert that into an int instead of raising an error? Might be tricky for non integer float edge case, e.g. 1.1